src/tqec/circuit/detectors/utils_test.py

```diff
@@ -98,19 +105,20 @@ def test_iter_by_moment_repeat_block() -> None:
             SHIFT_COORDS(0, 1)
             DETECTOR(1, 0) rec[-2] rec[-4]
             DETECTOR(3, 0) rec[-1] rec[-3]
-        }""")
+        }"""
+    )
     # REPEAT block should NOT be with the last MR before it, even without TICK.
     assert len(list(iter_stim_circuit_by_moments(circuit))) == 5
 
 
-def test_collapse_pauli_strings_at_moment_all_basis() -> None:
-    qubits = list(range(5))
+@pytest.mark.parametrize("basis", ["", "X", "Y", "Z"])
+@pytest.mark.parametrize("qubits", [range(5), [1, 2, 0], [0, 1, 2], [2, 1, 0]])
+def test_collapse_pauli_strings_at_moment_all_basis(basis, qubits) -> None:
```

Comment:
```suggestion
def test_collapse_pauli_strings_at_moment_all_basis(basis: typing.Literal["", "X", "Y", "Z"], qubits: typing.Iterable[int]) -> None:
```
should fix the typing issue (not tested locally)